public/app/app.ts

```diff
@@ -119,8 +119,8 @@ export class GrafanaApp {
 
       // Preload selected app plugins
       const promises: Array<Promise<any>> = [];
-      for (const modulePath of config.pluginsToPreload) {
-        promises.push(importPluginModule(modulePath));
+      for (const plugin of config.pluginsToPreload) {
+        promises.push(importPluginModule(plugin.path, plugin.version));
       }
```

Comment:
Can we take this opportunity and move this out of the main app boot so a failed plugin load does not cause all of Grafana to crash unrecoverably

Reply:
Yes! We were planning on doing this in a separate PR. Will link it shortly.

Reply:
@torkelo https://github.com/grafana/grafana/pull/41490

Reply:
:+1: